packages/lit/src/utils/range-manager.ts

```diff
@@ -192,6 +196,29 @@ export class RangeManager {
       `Cannot find virgo element in block ${point.path.join(' > ')}}`
     );
 
+    const maxLength = startVirgoElement.virgoEditor.yText.length;
```

Comment:
Can we extract the index and length calculation to another method?

Reply:
I will create other pr

Reply:
https://github.com/toeverything/blocksuite/pull/4129